README.md

```diff
@@ -1,21 +1,192 @@
 # ExJSONTemplate
 
-**TODO: Add description**
+ExJSONTemplate is an Elixir library which allows to write JSON templates in JSON.
 
-## Installation
+## JSONTemplate
 
-If [available in Hex](https://hex.pm/docs/publish), the package can be installed
-by adding `exjsontemplate` to your list of dependencies in `mix.exs`:
+The most basic template operation is string interpolation:
 
-```elixir
-def deps do
-  [
-    {:exjsontemplate, "~> 0.1.0"}
-  ]
-end
 ```
+in: {"user": "foo"}
+template: {"greet": "Hello {{ $.user }}"}
+out: {"greet": "Hello user"}
+```
+
+When using string interpolation scalar values (such as boolean, numbers, ...) are converted to string by default:
+
+```
+in: {"num": 42}
+template: {"answer": "The answer to the Ultimate Question of Life, the Universe, and Everything is {{ $.num }}"}
+out: {"answer": "The answer to the Ultimate Question of Life, the Universe, and Everything is 42"}
+```
+
+Beware that non scalar values (such as arrays) cannot be converted to string, therefore a rendering error is raised.
+
+String interpolation can be avoided using `{{{ }}}`, which is the preferred method for templating non string values:
+```
+in: {"number": 5}
+t: {"a": "{{{ $.number }}}"}
+out: {"a": 5}
+```
+
+`{{{ }}}` cannot be mixed with string interpolation, therefore `" {{{ $.test }}} a" causes a rendering error.
+
+### `&` (Unquote) Operator
 
-Documentation can be generated with [ExDoc](https://github.com/elixir-lang/ex_doc)
-and published on [HexDocs](https://hexdocs.pm). Once published, the docs can
-be found at [https://hexdocs.pm/exjsontemplate](https://hexdocs.pm/exjsontemplate).
+Unquote operator should be used when a scalar value is represented as string (such as `"5"`):
 
+```
+in: {"number": "5"}
+t: {"a": "{{& $.number}}"}
+out: {"a": 5}
+```
+
+If a string cannot be parsed as a scalar value, a rendering error is returned.
+
+When `&` operator is applied to any non string value, it behaves like `{{{ }}}`:
+
+```
+in: {"number": 5}
+t: {"a": "{{& $.number}}"}
+out: {"a": 5}
+```
+
+`&` can be also used for non scalar values such as:
+
+```
+in: {"nums": [0, 1, 2, 3, 4, 5]}
+t: {"nums2": ["{{& $.nums}}"]}
+out: {"nums2": [[0, 1, 2, 3, 4, 5]]}
+```
+
+`&` never renders strings, therefore it cannot be used for string interpolation.
+
+### `#` (Section) Operator
+
+Array templates can be written using `#` section operator:
+
+```
+in: {"repo": [{"name": "Davide"}, {"name": "Riccardo"}]}
+t: {"{{#repo}}": "Hello {{ $.name }}"}
+out: ["Hello Davide", "Hello Riccardo"]
+```
+
+```
+in: {"repo": ["Davide", "Riccardo"]}
+t: {"{{#repo}}": "Hello {{ $ }}"}
+out: ["Hello Davide", "Hello Riccardo"]
+```
+
+`#` operator can be also used for optional sections when applied to a truthy scalar:
```

Comment:
What is considered truthy? Would the example work also if `person` was `"foo"`? What if it was `0`?